WordPress/Classes/ViewRelated/Media/MediaLibraryViewController.swift

```diff
@@ -409,6 +447,56 @@ class MediaLibraryViewController: UIViewController {
         }
     }
 
+    // MARK: - Document Picker
+    
+    private func showDocumentPicker() {
+        let docTypes = [String(kUTTypeImage), String(kUTTypeAudiovisualContent)]
+        let docPicker = UIDocumentPickerViewController(documentTypes: docTypes, in: .import)
+        docPicker.delegate = self
+        self.present(docPicker, animated: true, completion: nil)
+    }
+    
+    func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentsAt urls: [URL]) {
+        showPreparingProgressHUD()
+        mediaProgressCoordinator.track(numberOfItems: urls.count)
+        // Wait until all assets are uploaded before we update the collection view
+        pickerDataSource.isPaused = true
+        
+        for documentURL in urls {
+            let mediaID = documentURL.lastPathComponent
```

Comment:
I'd be tempted to pull the contents of this loop out into another method – perhaps `makeAndUploadMediaWith(_ url: URL)` to mirror the one we currently have for assets. Also, the actual media creation / upload code is almost identical – I think perhaps they could be combined, and use the same completion block for the `service.createMedia` call maybe?